gpii/node_modules/processReporter/src/ProcessReporter.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * GPII Process Reporter
+ *
+ * Copyright 2015 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.require("kettle", require);
+fluid.require("./processesBridge.js", require);
+
+fluid.defaults("gpii.processReporter", {
+    gradeNames: "fluid.component"
+});
+
+fluid.defaults("gpii.processReporter.base", {
+    gradeNames: ["kettle.app", "gpii.processReporter"],
+    requestHandlers: {
+        processGet: {
+            route: "/processes",
+            type: "gpii.processReporter.allSolutions.get",
+            method: "get"
+        },
+        processGetSingle: {
+            route: "/processes/:solutionID",
+            type: "gpii.processReporter.solution.get",
+            method: "get"
+        }
+    },
```

Comment:
I think that is correct. There was a lot of inertia behind our "expose everything over HTTP REST" drive when we first worked on the framework, but I think we are now a lot clearer about our deployment use cases (described in @cindyli 's architecture diagrams) and there is no need to expose anything over REST that is only ever expected to be accessed via the local FlowManager. In time, we will eliminate almost all of our existing REST endpoints (minus the solutions registry and preferences server) in favour of electron-IPC channels